### PR TITLE
Rsync API cleanup

### PR DESCRIPTION
**Pull request by gabrielelanaro**
- Added docstrings and reference in the API docs
- dest->other in Tree.sync
- added checksum option
- fixed compress options
- Tree.sync works only with Tree not with Veg

### CODE REVIEW

File: docs/api_rsync.rst [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Treant syncronization
```
Comment (dotsdl) [inline docs/api_rsync.rst]: "syncronization" -> "synchronization"

File: src/datreant/core/rsync.py [hunk 2]
```diff
@@ -4,11 +4,44 @@
...
+    ----------
+    source : str
+        Source directory for the sync
+    dest : str
+        Dest directory for the sync
+    compress : bool
+        If True, use gzip compression to reduce the data transfered over
+        the network
+    backup : bool
+        If True, pre-existing files are renamed with a ~ extension before they
```
Comment (dotsdl) [inline src/datreant/core/rsync.py]: Nitpick: put "~" in quotes

File: src/datreant/core/trees.py [hunk 3]
```diff
@@ -469,25 +469,36 @@ def make(self):
...
         """Syncronize directories using rsync.
 
         Parameters
         ----------
-        dest: destination
+        other: str or Tree
+            Other end of the sync, can be either a path or another Tree.
+        mode: str
+            Either ``"upload"`` if uploading to  *other*, or ``"download"`` if
+            downloading from *other*
```
Comment (dotsdl) [inline src/datreant/core/trees.py]: Use single backticks for `other` here, since it's an arg name; this is one of the numpy doc conventions

**Comment by dotsdl** [comment]
Awesome work! Only minor comments/nitpicks. Ready to merge after these are addressed.

**Comment by gabrielelanaro** [comment]
I made a few corrections, let me know if I missed something

File: src/datreant/core/rsync.py [hunk 4]
```diff
@@ -4,11 +4,44 @@
...
+    ----------
+    source : str
+        Source directory for the sync
+    dest : str
+        Dest directory for the sync
+    compress : bool
+        If True, use gzip compression to reduce the data transfered over
+        the network
+    backup : bool
+        If True, pre-existing files are renamed with a "~" extension before they
```
Comment (dotsdl) [inline src/datreant/core/rsync.py]: We get a Travis failure from PEP8 checks that this line is too long now.

File: docs/trees.rst [hunk 5]
```diff
@@ -66,6 +66,22 @@ in which case whether a Tree or Leaf is returned is dependent on an ending
           sensitive to ending ``/`` separators to determine whether to give a
           Tree or a Leaf.
 
+Synchronizing Trees
+==================
```
Comment (dotsdl) [inline docs/trees.rst]: I can fix the rst here; very, very minor

**Comment by dotsdl** [comment]
Almost done. Again some minor issues but then we're good to merge.

**Comment by dotsdl** [comment]
Excellent. Merging!

**Pull request merged**